testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -423,4 +439,38 @@ public void testSelectInformationSchemaColumns()
                         "('tables'), " +
                         "('views')");
     }
+
+    @Test
+    public void testRenameTableAcrossSchema()
+    {
+        if (!hasBehavior(SUPPORTS_RENAME_TABLE_ACROSS_SCHEMAS)) {
+            // must reference an existing table in ALTER TABLE since name is validated before calling renameTable
+            assertQueryFails("ALTER TABLE nation RENAME TO other_schema.yyyy", "This connector does not support renaming tables.*");
+            return;
+        }
+
+        if (!hasBehavior(SUPPORTS_CREATE_SCHEMA)) {
+            throw new AssertionError("Cannot test ALTER TABLE RENAME across schemas without CREATE SCHEMA, the test needs to be implemented in a connector-specific way");
+        }
+
+        if (!hasBehavior(SUPPORTS_CREATE_TABLE)) {
+            throw new AssertionError("Cannot test ALTER TABLE RENAME across schemas without CREATE TABLE, the test needs to be implemented in a connector-specific way");
+        }
+
+        String tableName = "test_rename_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT 123 x", 1);
+
+        String schemaName = "schema_" + randomTableSuffix();
+        assertUpdate("CREATE SCHEMA " + schemaName);
+
+        String renamedTable = schemaName + ".test_rename_new_" + randomTableSuffix();
+        assertUpdate("ALTER TABLE " + tableName + " RENAME TO " + renamedTable);
+        assertQuery("SELECT x FROM " + renamedTable, "VALUES 123");
+
+        assertUpdate("DROP TABLE " + renamedTable);
+        assertUpdate("DROP SCHEMA " + schemaName);
+
+        assertFalse(getQueryRunner().tableExists(getSession(), tableName));
```

Comment:
This should be checked just after rename, not at the end.

Reply:
Done!